homeassistant/components/anthemav/config_flow.py

```diff
@@ -0,0 +1,120 @@
+    DEFAULT_PORT,
+    DOMAIN,
+    EMPTY_MAC,
+    UNKNOWN_MODEL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+async def connect_device(user_input: dict[str, Any]):
+    """Connect to the AVR device."""
+
+    @callback
+    def async_anthemav_update_callback(message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.debug("Received update callback from AVR: %s", message)
+        if "IDN" in message:
+            user_input[CONF_MAC] = None
+        elif "IDM" in message:
+            user_input[CONF_MODEL] = None
+        if CONF_MAC in user_input and CONF_MODEL in user_input:
+            deviceinfo_received.set()
+
+    deviceinfo_received = asyncio.Event()
+    avr = await anthemav.Connection.create(
+        host=user_input[CONF_HOST],
+        port=user_input[CONF_PORT],
+        auto_reconnect=False,
+        update_callback=async_anthemav_update_callback,
+    )
+    await avr.reconnect()
+    await asyncio.wait_for(deviceinfo_received.wait(), 5)
+    return avr
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Anthem A/V Receivers."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        avr = None
+        try:
+            avr = await connect_device(user_input)
+        except OSError:
+            _LOGGER.error(
+                "Can't established connection to %s:%s",
+                user_input[CONF_HOST],
+                user_input[CONF_PORT],
+            )
+            errors["base"] = "cannot_connect"
+        except asyncio.TimeoutError:
+            errors["base"] = "cannot_receive_deviceinfo"
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.exception(err)
+            errors["base"] = "unknown"
+        else:
+            user_input[CONF_MAC] = format_mac(avr.protocol.macaddress)
+            user_input[CONF_MODEL] = avr.protocol.model
+            if (
+                user_input[CONF_MAC] == EMPTY_MAC
+                or user_input[CONF_MODEL] == UNKNOWN_MODEL
+            ):
+                _LOGGER.error("Invalid MacAddress or model")
+                errors["base"] = "cannot_receive_deviceinfo"
+            else:
+                await self.async_set_unique_id(user_input[CONF_MAC])
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
```

Comment:
Can we use the host as the name instead of asking for it?

Reply:
Unfortunately the host is the IP address, and we there isn't any relevant information from the host we can use for the name.

Reply:
One other question on the name... Since the name is optional and may be empty, it seems like we still want a fallback here, and  using the IP:PORT as the fallback seems reasonable?   However, I also see name is also used for display in the media player where ip would be a little odd. Would it make sense to make it required rather than optional instead?

Reply:
Why not use a default name and if the user wants to change it can be done on both the config entry and device after the integration has been setup.

Reply:
I already setup a default name as part of the default value in the schema. And it is automatically taking the default value if you don't define anything in the UI. I've tested it. But you think we still need to make sure we setup the default value again here?

Reply:
Essentially take out name from the schema and use DEFAULT_NAME all the way through so we don't ask for the name

Reply:
What about if the user want to setup a custom name? We just rename the device an entity after?

Reply:
The default is to only ask the users of must have and name is not such as it can be pre-defined.
I think people in general would not bother to rename the config entry and probably would rename the device which in turn automatically rename the entities attached so the effort is quite slim.

Reply:
Removed the name from the UI and added a default value instead. Still importing the actual value previously setup in configuration.yaml